classes/Address.php

```diff
@@ -440,14 +440,23 @@ public static function getCountryAndState($id_address)
      *
      * @return bool The address exists
      */
-    public static function addressExists($id_address)
+    public static function addressExists($id_address, $refreshCache = false): bool
     {
-        return (bool) Db::getInstance(_PS_USE_SQL_SLAVE_)->getValue(
-            'SELECT `id_address`
-            FROM ' . _DB_PREFIX_ . 'address a
-            WHERE a.`id_address` = ' . (int) $id_address,
-            false
-        );
+        if ($id_address == 0) {
+            return false;
+        }
+        $cacheKey = 'address_exists_' . $id_address;
+        if ($refreshCache || !Cache::isStored($cacheKey)) {
```

Comment:
If a cache is put here then must make sure that it is invalidated whenever an address is created.

Reply:
... or deleted

Reply:
Since adresses are soft deleted and can be inactive, shouldn't the real test be `select id_address from address where id_address = ... and deleted = 0 and active = 1` ?
If this is true, not only this patch is only fixing the symptom, but not the errors in the logic.